atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1463,32 +1465,39 @@ private <T> ListenableFuture<Collection<Map.Entry<Cell, T>>> getWithPostFilterin
 
     private <T> ListenableFuture<Collection<Map.Entry<Cell, T>>> getWithPostFilteringIterate(
             TableReference tableReference,
-            Map<Cell, Value> remainingResultsToPostFilter,
+            Map<Cell, Value> resultsToPostFilter,
             Collection<Map.Entry<Cell, T>> resultsAccumulator,
             Function<Value, T> transformer,
             AsyncKeyValueService asyncKeyValueService,
             AsyncTransactionService asyncTransactionService) {
-        if (remainingResultsToPostFilter.isEmpty()) {
-            getCounter(AtlasDbMetricNames.SNAPSHOT_TRANSACTION_CELLS_RETURNED, tableReference)
-                    .inc(resultsAccumulator.size());
-            return Futures.immediateFuture(resultsAccumulator);
-        }
-
         return Futures.transformAsync(
-                getWithPostFilteringInternal(
-                        tableReference,
-                        remainingResultsToPostFilter,
-                        resultsAccumulator,
-                        transformer,
-                        asyncKeyValueService,
-                        asyncTransactionService),
-                remaining -> getWithPostFilteringIterate(
-                        tableReference,
-                        remaining,
-                        resultsAccumulator,
-                        transformer,
-                        asyncKeyValueService,
-                        asyncTransactionService),
+                Futures.immediateFuture(resultsToPostFilter),
```

Comment:
I decided to keep this as a future rather than executing directly in thread; this is wrapped upstream to be in a future _anyways_, but, the method arguments claim we give back a future, so we should assume that this an async method.

Reply:
First approach I was a bit more heavy handed in unwinding some of the downstream functions futures, but realized it's probably too invasive at this stage.